### PR TITLE
Simplify cloud-init scripts

### PR DESCRIPTION
**Pull request by dav3r**
Thanks to @jsf9k, we are now aware of a much cleaner way to get cloud-init to execute python scripts.  This PR removes python code from YAML templates and puts it in standalone python files (which now can be checked by pre-commit as Python, rather than as YAML).

I also moved all of the cloud-init scripts into their own directory to tidy things up a bit and pulled in the latest upstream changes to `.pre-commit-config.yaml` from [skeleton-generic](https://github.com/cisagov/skeleton-generic).

@KyleEvers, you probably want to do something similar in [rva-terraform](https://github.com/cisagov/rva-terraform).

### CODE REVIEW

**Review by jsf9k — APPROVED** [review]
Glad to see the simplifications taking root.
:deciduous_tree:

**Review by felddy — APPROVED** [review]
A wonderful Friday PR.

File: guacamole_cloud_init.tf [hunk 1]
```diff
@@ -5,31 +5,38 @@ data "template_cloudinit_config" "guacamole_cloud_init_tasks" {
...
+        guac_gophish_connection_name = var.guac_gophish_connection_name
+        pca_private_domain           = local.pca_private_domain
+        sql_template_fullpath        = "/root/guacamole_connection_template.sql"
+    })
+  }
+
+  part {
+    content_type = "text/x-shellscript"
+    content = templatefile(
+      "${path.module}/cloud-init/render-guac-connection-template-sql.py", {
```
Comment (mcdonnnj) [inline guacamole_cloud_init.tf]: I would like some consistency between the naming for `write-guac-connection-sql-template.yml` and `render-guac-connection-template-sql.py` if possible.

**Review by mcdonnnj — CHANGES_REQUESTED** [review]
Some good work here. 👍I had one nitpick for you to look at though.

File: guacamole_cloud_init.tf (hunk 1 above)
Comment (dav3r) [reply guacamole_cloud_init.tf]: Good catch - fixed in https://github.com/cisagov/pca-terraform/pull/5/commits/c8573cb7a017867a4365716cb1c31cc5da18c508.

**Review by mcdonnnj — APPROVED** [review]
Nice work! ✅

**Pull request merged**